base/bitarray.jl

```diff
@@ -530,14 +530,15 @@ function BitArray{N}(A::Array{Bool,N}) where N
     l = length(B)
     l == 0 && return B
     copy_to_bitarray_chunks!(Bc, 1, A, 1, l)
-    return B
+    return B::BitArray{N}
 end
 
 reinterpret(::Type{Bool}, B::BitArray, dims::NTuple{N,Int}) where {N} = reinterpret(B, dims)
 reinterpret(B::BitArray, dims::NTuple{N,Int}) where {N} = reshape(B, dims)
 
 if nameof(@__MODULE__) === :Base  # avoid method overwrite
-(::Type{T})(x::T) where {T<:BitArray} = copy(x)
+(::Type{T})(x::T) where {T<:BitArray} = copy(x)::T
+BitArray{N}(x::BitArray) where {N} = copy(x)::BitArray{N}
```

Comment:
Looks like this method will always fail with a `TypeError`? (If `x::BitArray{N}`, the method above is hit, otherwise, the type assertion on `copy(x)` will fail.) So maybe just throw a descriptive error explicitly?

Reply:
Good point, I think we can just delete that method, it shouldn't ever be called.